### ISSUE TITLE
Possible bug on changing settings of a SPFx app teams tab 

### ISSUE THREAD

**Issue by canturan** [SharePoint/sp-dev-docs#9389]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

None

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

New Teams desktop client

### Describe the bug / error

As mentioned on this closed issue: [https://github.com/SharePoint/sp-dev-docs/issues/9253](https://github.com/SharePoint/sp-dev-docs/issues/9253), it looks like there is still a bug on the settings panel of a SPFx teams tab. 

When I try to edit the settings I got the "you added a new Tab-Dialog" after that the settings panel does not load the current settings but the initial settings. I can alter the settings and save it. The new setting are accepted and working but it is very confusing that the current settings get lost each time I click configure.

### Steps to reproduce

Edit the settings of a SPFx teams tab on the new Teams desktop client 

### Expected behavior

That the settings panel loads the correct current settings and not the initial settings

**Comment by kristofferarn** [comment]
Also facing the same issue. Tested on both older and newer versions of SPFx.

**Comment by sharepointalist** [comment]
We have the same issue. Effectively there is no way to edit settings for the SPFx app in Teams Tab.

**Comment by kandharithirumalreddy** [comment]
we have facing the issue. any update?

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

We have noticed that,  when we change webpart settings in new teams - but it seems that everytime you edit the webpart settings (of a existing tab), it generating a new webpart instance id.

FYI : Clearly there is something wrong with the URL replacement. Technically a new tab will be generated (because the tab-url is not available anymore).

I'm using spfx 1.18.2

**Comment by tpalacino** [comment]
@nick-pape Any update on this? Is it even on the radar of the New Teams developers?

I opened a Microsoft Support case and this was the response:

> Thanks for contacting Microsoft support regarding unable to update channel tab configurations.
> 
> As discussed over call, please use classic teams instead of New Teams. As confirmed during call you are able to classic teams without any issue. Please request to your client to use classic team until it will not be debugged.

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

**Comment by shaipetel** [comment]
FYI, I think this issue is a duplicate of this one: [9464](https://github.com/SharePoint/sp-dev-docs/issues/9464)

**Comment by antonkhrit** [comment]
It has been five months since the issue was reported and is still in "bug-suspected" status. 

@nick-pape  It is a bug. Please fix it.

@VesaJuvonen I saw you participated in the discussion for the duplicate issue: https://github.com/SharePoint/sp-dev-docs/issues/9464

Could you provide any updates or plans?

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us.

**Comment by tpalacino** [comment]
@AJIXuMuK I saw you resolved the #9253. Any chance you can get this issue in front of the right people? This has become a significant issue for many ISVs.

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us

**Comment by calechko** [comment]
> Hi,
> 
> any update on this issue, when will it be fixed? will the fix be part of next release?
> 
> Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us

Same here - our customers are becoming upset. We need to get an answer on this as soon as possible.

**Comment by tpalacino** [comment]
@andrewconnell have you experienced this issue on any of your SPFx teams tab projects?

**Comment by andrewconnell** [comment]
@tpalacino No, because I don't use SPFx for creating tabs for Microsft Teams. For various reasons, I think SPFx is a bad option and instead, building tabs using the native Teams AppDev approach is much better.

**Comment by tpalacino** [comment]
@andrewconnell Thanks for the tip. Do you know if Microsoft is planning on removing support for syncing SPFx apps to Teams?

**Comment by shaipetel** [comment]
I can confirm this is still an issue with using a dedicated teams manifest (manually created), but deployed through an SPFx package...
It is still a tab configured to load an SPFx web part inside a tab.

**Comment by andrewconnell** [comment]
@tpalacino said:
> Do you know if Microsoft is planning to remove support for syncing SPFx apps to Teams?

No, I have no knowledge of their future plans WRT SPFx+Teams. But... 

- the [recent announcement of the deprecation & pending retirement of isolated web parts](https://www.voitanos.io/blog/sharepoint-framework-domain-isolated-webpart-retirement/), presumably (_I've asked, but not had a response) because of their negative view of `iframe`s...
- ... and the fact I see more investments on new things rather than fixing long-standing issues or completing work/features/functionality that was promised long ago...

... I don't have the confidence nor recommend developers invest in using SPFx for extending Teams.

The benefits to using SPFx to create Teams tabs don't outweigh the downsides of doing so, including the fact you can't do anywhere near as much stuff as you can if you're doing native Teams App Dev (plus, the dev toolchain & development experience is better less prohibitive). [I've written extensibility about my feelings on this topic](https://www.voitanos.io/blog/microsoft-365-apps-use-spfx-or-microsoft-teams/).

**Comment by juliemturner** [comment]
> The benefits to using SPFx to create Teams tabs don't outweigh the downsides of doing so

I have to disagree with @andrewconnell somewhat in the sense that if your solution needs to be surfaced in both SharePoint and Teams this is a LOT of duplication. That said, I think the work around is to build your solution completely as a library (NOT SPFx library) and then essentially bootstrap it into an SPFx solution and then surface it in a Teams tab by hosting the UI in an azure resource.

(Note I don't think I would advocate for surfacing your hosted solution in an iFrame in an SPFx web part simply because the SharePoint pages team does not style iFrames appropriately and therefore your responsive experience will be poor at best.)

**Comment by tpalacino** [comment]
Thanks for sharing that perspective @juliemturner. I haven't personally used the native Teams AppDev because I was concerned about the level of reusability. Seems like I've got some research to do.

**Comment by shaipetel** [comment]
My 2 cents on this, I'm with @juliemturner on this - the iframe experience is not always ideal.
There are a lot more downsides to it that I think @andrewconnell had mentioned (in some scenarios, IMHO)

- It requires a backend even if your app simply access information that's already on the site.
- Styling is much more difficult.
- SSO authentication
- Performance hit, vs a pure client side application (if your solution allows it)

Having your SPFx exposed as a teams tab loads it under your SharePoint domain, which allows you seamless SharePoint REST API access - which decreases dev time/complexity significantly on both your SharePoint app side and Teams.

That opens the door for having your custom solutions store all information directly on the user's tenant (the connected site, in lists and libraries), never having this information even pass through your servers - which is a huge advantage for many customers who are sensitive about where their information is transmitted / stored.

There are many more advantages in my opinion but its missing the point of this thread.

Bottom line, this is a matter of opinion, the facts here are there is a **breaking change**, a bug in the new teams, and I think we can all agree it is **very concerning** that MSFT aren't looking into resolving it.

Its damaging all of our reputation and putting us in a bad position between MSFT and our customers, and this situation is not ideal. All I can do is point customers to this discussion, since if they open a support ticket - the answer is "oh, its a custom solution? talk to the vendor"

**Comment by andrewconnell** [comment]
@shaipetel To be fair, I've got a long list of downsides, but I don't want to hijack this issue more than we already have so I'll refrain from doing the back and forth here :)

**Comment by tpalacino** [comment]
We identified a workaround. [demo video](https://www.youtube.com/watch?v=jA4V-SU9GLw)

**Comment by antonkhrit** [comment]
Hey guys! 

The fact is that there is an ability to use SPFx as a tab. This feature worked correctly for a long time. Now it doesn't. Clients are suffering. Developers don't have a way to fix this. 

Microsoft must provide backward compatibility for things like this.

I don't think developing a solution for Teams from scratch is an answer here. It depends on the solution itself. In some cases, it is a significant investment. It should not be dictated by a lack of backward compatibility in Microsoft Teams client. 

This is a bug that must be fixed. Other discussions around best practices and IFrames are not justifying the lack of backward compatibility.

It is almost seven months since the bug was reported with no response from Microsoft whatsoever.

**Comment by kandharithirumalreddy** [comment]
Hi,

any update on this issue, when will it be fixed? will the fix be part of next release?

Now Microsoft forcing to use New Teams. So it's effecting our customers. Now It's become major issue for us

I cannot understand why they are unable to fix this issue. The issue is this always they are creating a new Instance for the webpart properties, instead if the maintain some kind of ID/TabUrl/WebpartInstanceId then it will update if already added.

**Comment by chr-sad** [comment]
Hi,

is it only on our tenant that now in New Teams (at least if we open Teams in the Browser) that if we want to access and edit the settings in a Teams-SPFx-Tab which was already created, that the pane not even opens any more, but only if the SPFx Tab is added for the first time?

Is there any news on this issue? What is the "official" message from Microsoft - what to tell customers, who want to use (and already used) this feature?

**Comment by singingknight** [comment]
> Hi,
> 
> is it only on our tenant that now in New Teams (at least if we open Teams in the Browser) that if we want to access and edit the settings in a Teams-SPFx-Tab which was already created, that the pane not even opens any more, but only if the SPFx Tab is added for the first time?
> 
> Is there any news on this issue? What is the "official" message from Microsoft - what to tell customers, who want to use (and already used) this feature?

See this related Issue https://github.com/SharePoint/sp-dev-docs/issues/9464#issuecomment-2299322443

**Comment by Ashlesha-MSFT** [comment]
@canturan, @singingknight,
Thank you for bringing this issue to our attention. Could you please confirm if the issue still persists for you?

**Comment by singingknight** [comment]
> [@canturan](https://github.com/canturan), [@singingknight](https://github.com/singingknight), Thank you for bringing this issue to our attention. Could you please confirm if the issue still persists for you?

#9464 Solved it for me and works stable now @Ashlesha-MSFT

**Comment by Ashlesha-MSFT** [comment]
@tpalacino, @calechko, @shaipetel, @kandharithirumalreddy,
Could you please confirm if the issue is now resolved on your end as well? 
We’d like to proceed with closing this issue, but want to ensure it’s fully resolved for everyone first. 
Thanks!

**Comment by tpalacino** [comment]
> [@tpalacino](https://github.com/tpalacino), [@calechko](https://github.com/calechko), [@shaipetel](https://github.com/shaipetel), [@kandharithirumalreddy](https://github.com/kandharithirumalreddy), Could you please confirm if the issue is now resolved on your end as well? We’d like to proceed with closing this issue, but want to ensure it’s fully resolved for everyone first. Thanks!

@Ashlesha-MSFT Confirmed settings are back to working properly. Client: 

<img width="1306" height="943" alt="Image" src="https://github.com/user-attachments/assets/0dc839ed-6e42-426c-932e-23d855cd3e18" />

**Comment by Ashlesha-MSFT** [comment]
Based on the recent confirmations, the issue with changing settings of SPFx Teams tabs in the new Teams client has been resolved.

Since there are no further reports of this problem persisting, we will proceed with closing this issue. If anyone encounters the problem again, please feel free to reopen or create a new issue.
